app/src/main/java/com/portfolio/tasky/entry/usecases/domain/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.portfolio.tasky.entry.usecases.domain
+
+import android.content.Context
+import com.google.gson.Gson
+import com.portfolio.tasky.R
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import javax.inject.Inject
+
+class UserPreferencesImpl @Inject constructor(val context: Context) : UserPreferences {
+
+    private val sharedPreference by lazy { context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)}
```

Comment:
Since you are saving sensitive information, like a user token, I recommend using an EcryptedSharedPreference

Reply:
for current implementation I'll be keeping this and in a future PR I'll be storing access_token in keystore 👍